plugins/processors/ifname/ifname.go

```diff
@@ -132,6 +139,16 @@ func (d *IfName) Init() error {
 		return fmt.Errorf("parsing SNMP client config: %w", err)
 	}
 
+	if d.Translator == "gosmi" {
+		var err error
+		d.translator, err = si.NewGosmiTranslator(d.Path, d.Log)
+		if err != nil {
+			return err
+		}
+	} else {
+		d.translator = si.NewNetsnmpTranslator()
+	}
+
```

Comment:
This doesn't make much sense, since I made the plugin independent from any MIBs.

Reply:
Although it's independent from MIBS because it hard codes numeric OIDs, the table and field initialization code still needs a translator. Inside those init functions, I don't expect any translation to happen. I did change ifname's code so it doesn't switch between translators.

Reply:
You made the init functions depenant on a translator interface in this PR, so it can still be fixed.